app/src/main/java/com/orange/ods/app/ui/search/SearchScreen.kt

```diff
@@ -16,41 +16,157 @@ import androidx.compose.foundation.lazy.LazyColumn
 import androidx.compose.foundation.lazy.items
 import androidx.compose.runtime.Composable
 import androidx.compose.runtime.MutableState
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
 import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.painter.ColorPainter
+import androidx.compose.ui.platform.LocalContext
 import androidx.compose.ui.res.painterResource
 import androidx.compose.ui.res.stringResource
 import androidx.compose.ui.text.input.TextFieldValue
+import com.orange.ods.app.R
+import com.orange.ods.app.ui.LocalOdsGuideline
+import com.orange.ods.app.ui.components.Component
+import com.orange.ods.app.ui.components.Variant
 import com.orange.ods.app.ui.components.components
+import com.orange.ods.app.ui.guidelines.color.DialogColor
+import com.orange.ods.app.ui.guidelines.color.getValue
+import com.orange.ods.app.ui.guidelines.spacing.Spacing
+import com.orange.ods.app.ui.utilities.DrawableManager
 import com.orange.ods.compose.component.list.OdsListItem
 import com.orange.ods.compose.component.list.OdsListItemIcon
 import com.orange.ods.compose.component.list.OdsListItemIconType
 import com.orange.ods.compose.component.list.iconType
+import com.orange.ods.theme.guideline.GuidelineColor
+import com.orange.ods.theme.guideline.toHexString
 import com.orange.ods.utilities.extension.orElse
 
 @Composable
-fun SearchScreen(searchedText: MutableState<TextFieldValue>, onComponentClick: (Long) -> Unit) {
-    val filterComponents = components.filter { component ->
-        searchedText.value.text.isEmpty() || stringResource(id = component.titleRes).lowercase().contains(searchedText.value.text.lowercase())
+fun SearchScreen(
+    searchedText: MutableState<TextFieldValue>,
+    onComponentClick: (Long) -> Unit,
+    onVariantClick: (Long) -> Unit,
+    onGuidelineSpacingClick: (Long) -> Unit
```

Comment:
Here we can replace the 3 parameters in only one:
```kotlin
onResultItemClick: (String, Long?) -> Unit
```

When done, we can display SearchScreen with this parameter:
```kotlin
onResultItemClick = { route, id -> navigateToElement(route, id, from) }
```

And in the `when`:
```kotlin
is Component -> onResultItemClick(MainDestinations.ComponentDetailRoute, item.id)
is Variant -> onResultItemClick(MainDestinations.ComponentVariantRoute, item.id)
is Spacing -> onResultItemClick(MainDestinations.GuidelineSpacing, null)
```